### PR TITLE
Improving types

### LINKED ISSUE #3: Improve typing
Improve lib types based on TODOs placed

### PR DESCRIPTION
**Pull request by SampaioLeal**
Some generic types in methods to improve developer experience
This will close #3

### CODE REVIEW

File: src/FormixStore.ts
Comment (Viniciusvcr) [inline src/FormixStore.ts]: for all the get* functions, i think tou should define and follow one style: explicitly mark the function's return type or using "as type"

File: src/useField.ts
Comment (Viniciusvcr) [inline src/useField.ts]: make optional for the user to specify the element type

```suggestion
export default function useField<T, Element extends HTMLInputElement = HTMLInputElement>(
```

**Review by Viniciusvcr — CHANGES_REQUESTED** [review]

File: src/utils.ts
Comment (devzgabriel) [inline src/utils.ts:74]: Why do you use `any` at the set function and `unknown` at the createNewElement function?

```ts
let vAny: any = 10;          // We can assign anthing to any
let vUnknown: unknown =  10; // We can assign anthing to unknown just like any 


let s1: string = vAny;     // Any is assigable to anything 
let s2: string = vUnknown; // Invalid we can't assign vUnknown to any other type (without an explicit assertion)
```

**Review by devzgabriel — APPROVED** [review]

File: src/utils.ts
Comment (SampaioLeal) [reply src/utils.ts:74]: At first i prefer to use `unknown`, but it may be a headache on some expressions like `object[acessor] = newValue`
so, to avoid some rage moments with typescript i use `any`

Comment (SampaioLeal) [reply src/utils.ts:74]: can i show you the problems later?

Comment (devzgabriel) [reply src/utils.ts:74]: Sure, I approved the PR. We can talk about this later.
I'm Just waiting for Vinicius's approval to make the merge.

Comment (SampaioLeal) [reply src/utils.ts:74]: I have to work on some requested changes.
After approval i can text you :)

File: src/FormixStore.ts
Comment (SampaioLeal) [reply src/FormixStore.ts]: done :)

**Review by Viniciusvcr — APPROVED** [review]

**Pull request merged**
